### PR TITLE
3.12 version OHOS adaptation

### PR DESCRIPTION
**Pull request by wanghui187**

### CODE REVIEW

File: cocos/renderer/CCGLProgramState.cpp [hunk 1]
```diff
@@ -362,8 +362,11 @@ GLProgramState::GLProgramState()
 , _vertexAttribsFlags(0)
 , _glprogram(nullptr)
 , _nodeBinding(nullptr)
+#if (CC_TARGET_PLATFORM == CC_PLATFORM_OHOS)
+, _backToForegroundlistener(nullptr)
+#endif
```
Comment (minggo) [inline cocos/renderer/CCGLProgramState.cpp]: Why only initialize on OHOS?

File: cocos/renderer/CCMeshCommand.cpp [hunk 2]
```diff
@@ -54,10 +54,13 @@ MeshCommand::MeshCommand()
 , _vao(0)
 , _material(nullptr)
 , _stateBlock(nullptr)
+#if (CC_TARGET_PLATFORM == CC_PLATFORM_OHOS)
+, _rendererRecreatedListener(nullptr)
+#endif
```
Comment (minggo) [inline cocos/renderer/CCMeshCommand.cpp]: Why only initialize on OHOS?

File: cocos/scripting/lua-bindings/CMakeLists.txt [hunk 3]
```diff
@@ -123,12 +148,39 @@ if(MACOSX)
...
 endif()
 
 set(lua_bindings_files ${lua_cocos2d_source_files} ${lua_bindings_manual_files} ${lua_bindings_auto_files})
 
 add_library(luacocos2d ${lua_bindings_files})
 target_link_libraries(luacocos2d cocos2d)
+
+if(OHOS)
+    target_link_libraries(luacocos2d ext_luajit)
+    # luajit比lua多一个头文件luajit.h，在../../../external/lua/luajit/src下，暂时没有用到
```
Comment (minggo) [inline cocos/scripting/lua-bindings/CMakeLists.txt]: Please don't use Chinese. Please check all files.

File: cocos/scripting/lua-bindings/script/framework/device.lua [hunk 4]
```diff
@@ -58,6 +58,8 @@ elseif target == cc.PLATFORM_OS_WINRT then
     device.platform = "winrt"
 elseif target == cc.PLATFORM_OS_WP8 then
     device.platform = "wp8"
+elseif target == cc.PLATFORM_OS_OHOS then
+    device.platform = "harmony"
```
Comment (minggo) [inline cocos/scripting/lua-bindings/script/framework/device.lua]: Should change to `HarmonyOS Next`?

File: cocos/renderer/CCGLProgramState.cpp (hunk 1 above)
Comment (wanghui187) [reply cocos/renderer/CCGLProgramState.cpp]: fixed

File: cocos/renderer/CCMeshCommand.cpp (hunk 2 above)
Comment (wanghui187) [reply cocos/renderer/CCMeshCommand.cpp]: fixed

File: cocos/scripting/lua-bindings/CMakeLists.txt (hunk 3 above)
Comment (wanghui187) [reply cocos/scripting/lua-bindings/CMakeLists.txt]: fixed

File: cocos/scripting/lua-bindings/script/framework/device.lua (hunk 4 above)
Comment (wanghui187) [reply cocos/scripting/lua-bindings/script/framework/device.lua]: fixed

**Pull request merged**